db/structure.sql

```diff
@@ -1052,7 +1052,8 @@ CREATE TABLE public.invoices (
     cancelled_at timestamp without time zone,
     total numeric(10,2) NOT NULL,
     in_directo boolean DEFAULT false,
-    buyer_vat_no character varying
+    buyer_vat_no character varying,
+    date date
```

Comment:
I am not sure if this is the right choice. 

You need to know that it means the date the invoice was issued, and not due date, and it conflicts with SQL `DATE` keyword.

Reply:
>I agree that DATE itself is a reserved keyword, but in what way and what exactly conflicts?

It is a reserved keyword in all [3 recent SQL standards](https://www.postgresql.org/docs/current/static/sql-keywords-appendix.html), which makes it generally not portable. In Oracle you cannot query such a column without escaping its name, it raises a parser exception.

> I didn't notice any obstacles using it. Even if you need to use plain SQL, you just need to quote it.

That is precisely one of the reasons why you should avoid doing it. Quoted values are case sensitive, making them prone to bugs caused by typos and key rollovers.

```sql
create table dates(DATE DATE NOT NULL, ISSUE_DATE DATE NOT NULL);

select * from dates;
+--------+--------------+
| date   | issue_date   |
|--------+--------------|
+--------+--------------+

SELECT "Date" from dates;
column "Date" does not exist
LINE 1: SELECT "Date" from dates
               ^
HINT:  Perhaps you meant to reference the column "dates.date".

SELECT ISSUE_Date from dates;
+--------------+
| issue_date   |
|--------------|
+--------------+
SELECT 0
```

> Invoice has date so date seems like a natural, self-explanatory name of a column.

It has 3 dates to be exact: the date when it was created, the date when it was sent to the customer and the date when it is due. Because of that, just date is not the best choice IMHO.

Reply:
OK, at least now I understand the point. However, I still see `date` as a better fit. I bet nobody will ask you to clarify, if you say just "Invoice date".

Reply:
@artur-beljajev please fix the reserved date keyword issue. You always fight against ambiguity so it is hard to understand why all of the sudden its OK even if it is error prone and raises specific problems as @maciej-szlosarczyk described.

Reply:
I don't see a problem using `date`, which is a standard term for invoice.
https://www.google.ee/search?hl=en&as_q=&as_epq=invoice+date&as_oq=&as_eq=due&as_nlo=&as_nhi=&lr=&cr=&as_qdr=all&as_sitesearch=&as_occt=any&safe=images&as_filetype=&as_rights=
Omniva uses the same term: https://media.voog.com/0000/0042/1620/files/Estonian_e-invoice_description_ver1.2_eng.pdf
Our `estonian_e_invoice` gem also sticks to it.

I find `invoice.date` clearer. 99.9% of time we will use application layer (i.e. call `invoice.date`), instead of manually building queries (in which case you will need to quote it). I don't find `invoice.date` ambiguous.

Please elaborate, what problem it introduces in our case?

Reply:
P.S. Another solution is to employ `alias_attribute :date, :issue_date`